packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx

```diff
@@ -0,0 +1,81 @@
+import { BB_SY_CREAM, BB_SY_AAVE, BB_SY_COMPOUND, APP_URL, SY_URL } from '../constants'
+import type { SYCoinProps } from '../UI/SmartYieldPoolView'
+import { useAsyncRetry } from 'react-use'
+import urlcat from 'urlcat'
+
+export type SYPoolModelData = { [id: string]: SYCoinProps[] }
+
+export function SmartYieldPoolModelGetData() {
+    return useAsyncRetry(async () => {
+        const response = await fetch('https://api.barnbridge.com/api/smartyield/pools?originator=all', {
```

Comment:
Move this external API request to the worker module. The SNS provider may enable the CSP, which means external requests are not allowed.

Reply:
Interesting -- any examples of what the worker module looks like; sorry i'm not too familiar with it

Reply:
Please ref to the Gitcoin plugin as an example. You can find it here in `packages/maskbook/src/plugins/Gitcoin/index.ts`.

Reply:
I've spent a few hours on this and I can't figure it out. My objects never are populated; I need some help understanding what I'm doing wrong. Sending out a push